src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,204 @@
+            ArrayList<String> texts = parser.convertTaskInput(input);
+            return new TodoCommand(texts);
+
+        } else if (input.contains(command.delete.toString())) {
+            int selected = Integer.parseInt(input.substring(input.indexOf("delete") + 7)) - 1;
+            return new DeleteCommand(selected);
+        } else if (input.contains(command.find.toString())) {
+            if (input.equals(command.find.toString())) {
+                throw new DukeException("find cannot be empty");
+            }
+            if (input.charAt(4) != ' ') {
+                throw new DukeException("Invalid command");
+            }
+            Parser parser = new Parser("P");
+            ArrayList<String> texts = parser.convertTaskInput(input);
+            return new FindCommand(texts.get(0));
+        } else {
+            throw new DukeException("Invalid command");
+        }
+    }
+
+    /**
+     * Converts the user input of different type of tasks into ArrayList.
+     * @param input User input of the tasks.
+     * @return Required data of task from the user input in ArrayList.
+     * @throws DukeException If user typed invalid tasks commands.
+     */
+    public ArrayList<String> convertTaskInput(String input) throws DukeException {
+        ArrayList<String> texts = new ArrayList<>();
+        switch (type) {
+        case "E": {
+            if (input.indexOf("/by") - 1 < 6) {
+                throw new DukeException("Your Event name is blank");
+            }
+            String desc = input.substring(6, input.indexOf("/from") - 1);
+            if (desc.isBlank() || desc.isEmpty()) {
+                throw new DukeException("Your Event name is blank");
+            }
+            texts.add(desc);
+            String from = input.substring(input.indexOf("/from") + 6, input.indexOf("/to") - 1);
+            texts.add((from));
+            String to = input.substring(input.indexOf("/to") + 4);
+            texts.add(to);
+            break;
+        }
+        case "D": {
+            if (input.indexOf("/by") - 1 < 9) {
+                throw new DukeException("Your Deadline name is blank");
+            }
+            String desc = input.substring(9, input.indexOf("/by") - 1);
+            if (desc.isBlank() || desc.isEmpty()) {
+                throw new DukeException("Your Deadline name is blank");
+            }
+            texts.add(desc);
+            String by = input.substring(input.indexOf("/by") + 4);
+            texts.add(by);
+            break;
+        }
+        case "T": {
+            String desc = input.substring(input.indexOf("todo") + 5);
+            if (desc.isBlank() || desc.isEmpty()) {
+                throw new DukeException("Your Todo name is blank");
+            }
+            texts.add(desc);
+            break;
+        }
+        case "P": {
+            String keyWord = splitter(input, " ").get(1);
+            texts.add(keyWord);
+            break;
+        }
```

Comment:
I like the good use of switch statements here, as well as its proper formatting and indentation